cli/cmd/agent_installctf.go

```diff
@@ -0,0 +1,254 @@
+	runners, err := awsFindRunnersToCapture()
+	if err != nil {
+		return err
+	}
+
+	for _, runner := range runners {
+		cli.Log.Debugw("runner: ", "runner", runner)
+		cli.Log.Debugw("runner user: ", "user", runner.Runner.User)
+		cli.Log.Debugw("runner region: ", "region", runner.Region)
+		cli.Log.Debugw("runner az: ", "az", runner.AvailabilityZone)
+		cli.Log.Debugw("runner instance ID: ", "instance ID", runner.InstanceID)
+		cli.Log.Debugw("runner: ", "runner hostname", runner.Runner.Hostname)
+		err := runner.SendAndUseIdentityFile()
+		if err != nil {
+			cli.Log.Debugw("ec2ic failed", "err", err)
+			continue
+		}
+
+		if err := verifyAccessToRemoteHost(&runner.Runner); err != nil {
+			cli.Log.Debugw("verifyAccessToRemoteHost failed")
+			return err
+		}
+
+		if alreadyInstalled := isAgentInstalledOnRemoteHost(&runner.Runner); alreadyInstalled != nil {
+			cli.Log.Debugw("agent already installed on host, skipping")
+			continue
+		}
+
+		token := agentCmdState.InstallAgentToken
+		if token == "" {
+			// user didn't provide an agent token
+			cli.Log.Debugw("agent token not provided")
+			var err error
+			token, err = selectAgentAccessToken()
+			if err != nil {
+				return err
+			}
+		}
+		cmd := fmt.Sprintf("sudo sh -c \"curl -sSL %s | sh -s -- %s\"", agentInstallDownloadURL, token)
+		err = runInstallCommandOnRemoteHost(&runner.Runner, cmd)
+		if err != nil {
+			cli.Log.Debugw("runInstallCommandOnRemoteHost failed")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func awsFindRunnersToCapture() ([]*lwrunner.AWSRunner, error) {
+	regions, err := awsFindRegions()
+	if err != nil {
+		return nil, err
+	}
+
+	allRunners := []*lwrunner.AWSRunner{}
+	for _, region := range regions {
+		regionRunners, err := awsFindRunnersInRegion(*region.RegionName)
+		if err != nil {
+			return nil, err
+		}
+		allRunners = append(allRunners, regionRunners...)
+	}
+
+	return allRunners, nil
+}
+
+// awsFindRegions queries the AWS API to list all the regions that
+// are enabled for the user's AWS account. Use the "include_regions"
+// command-line flag to only get regions in this list.
+func awsFindRegions() ([]types.Region, error) {
+	// Describe all regions that are enabled for the account
+	var filters []types.Filter
+	if len(agentCmdState.CTFIncludeRegions) > 0 {
+		filters = []types.Filter{
+			{
+				Name:   aws.String("region-name"),
+				Values: agentCmdState.CTFIncludeRegions,
+			},
+		}
+	}
+	input := &ec2.DescribeRegionsInput{
+		Filters: filters,
+	}
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	svc := ec2.New(ec2.Options{
+		Credentials: cfg.Credentials,
+		Region:      "us-west-2", // use us-west-2 for lack of a better region
+	})
+
+	output, err := svc.DescribeRegions(context.TODO(), input)
+	if err != nil {
+		return nil, err
+	}
+	return output.Regions, nil
+}
+
+func awsFindRunnersInRegion(region string) ([]*lwrunner.AWSRunner, error) {
```

Comment:
Rather than inventing our own terminology, I think we might prefer to use the AWS terminology:

```suggestion
func awsRegionDescribeInstances(region string) ([]*lwrunner.AWSRunner, error) {
```